### PR TITLE
First commit.  This fork adds QA tests...

### PR DESCRIPTION
**Pull request by jackpr-unity**
…, which right now consist of manual sanity check scenes meant to accelerate development.

- Keyboard and mouse scene
- Start at Xbox XInput scene.  Leaving this right now to pivot to touch testing.
- Updated .gitignore to include a Builds folder to enable building for tests.

### CODE REVIEW

File: Assets/QA/Tests/KeyboardMouseBasic/Scripts/ButtonStateHEX.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+using UnityEngine.UI;
+using ISX;
+
+
+[RequireComponent (typeof(Text))]
+public class ButtonStateHEX : MonoBehaviour {
```
Comment (Rene-Damm) [inline Assets/QA/Tests/KeyboardMouseBasic/Scripts/ButtonStateHEX.cs]: Please run Unity formatting tools on all C# code before committing. On OSX, run format.sh from the repo root. On Windows, run format.ps1 from there.

File: Assets/QA/Tests/KeyboardMouseBasic/Scripts/ButtonStateHEX.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    [Header("This text box should have a monospace font")]
+    Text OutputText;
```
Comment (Rene-Damm) [inline Assets/QA/Tests/KeyboardMouseBasic/Scripts/ButtonStateHEX.cs]: Please conform to Unity code conventions for all C# code in the repo.

File: Assets/QA/Tests/KeyboardMouseBasic/Scripts/MouseOutputs.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    public Text PositionY;
+    public Text DeltaX;
+    public Text DeltaY;
+    public Text ScrollX;
+    public Text ScrollY;
+
+    // Use this for initialization
+    void Start () {
+		
+	}
```
Comment (Rene-Damm) [inline Assets/QA/Tests/KeyboardMouseBasic/Scripts/MouseOutputs.cs]: Remove.

File: Assets/QA/Tests/KeyboardMouseBasic/Scripts/KeyboardLastKey.cs [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    {
+        if (KB != null && Keyboard.current != null && KB != Keyboard.current)
+        {
+            ReleaseKB();
+            SetCurrentKB();
+        }
+        else if (KB == null){
+            SetCurrentKB();
+        }
+        //if (KB.)
```
Comment (Rene-Damm) [inline Assets/QA/Tests/KeyboardMouseBasic/Scripts/KeyboardLastKey.cs]: Please clean up commented out code.

**Review by Rene-Damm — CHANGES_REQUESTED** [review]

File: Assets/QA/Tests/KeyboardMouseBasic/Scripts/ButtonStateHEX.cs (hunk 2 above)
Comment (jackpr-unity) [reply Assets/QA/Tests/KeyboardMouseBasic/Scripts/ButtonStateHEX.cs]: Specifically, you are referring to the script name in this instance?

Comment (Rene-Damm) [reply Assets/QA/Tests/KeyboardMouseBasic/Scripts/ButtonStateHEX.cs]: To several places. Private property/field names need to be camelCase. Public ones m_PascalCase. Local variables camelCase. No _ prefixes on methods. Probably a couple more things.

**Comment by jackpr-unity** [comment]
New commit that adheres to coding conventions.

**Comment by Rene-Damm** [comment]
The files in .vs/ seem like accidental commits.

**Comment by Rene-Damm** [comment]
Other than the .vs files, LGTM.

**Comment by jackpr-unity** [comment]
Yep, didn't realize that I had to manually remove files that were already tracked after changing the .gitignore

**Pull request merged**